src/perform_step/bdf_perform_step.jl

```diff
@@ -454,3 +454,131 @@ function perform_step!(integrator,cache::QNDF2Cache,repeat_step=false)
   cache.newton_iters = iter
   f(integrator.fsallast, u, p, t+dt)
 end
+
+function initialize!(integrator, cache::QNDFConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+function perform_step!(integrator,cache::QNDFConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack udiff,dts,k,max_order,D,D2,R,U = cache
+  cnt = integrator.iter
+  κ = integrator.alg.kappa[k]
+  γ = inv((1-κ)*γₖ[k])
+  flag = true
+  for i in 2:k
+    if dts[i] != dts[1]
+      flag = false
+      break
+    end
+  end
+  if cnt > 2
+    if flag
+      ρ = dt/dts[1]
+      # backward diff
+      backward_diff(udiff,D,D2,k)
+      if ρ != 1
+        U!(k,U)
+        R!(k,ρ,cache)
+        D .= D * (R * U)
+      end
+    else
+      for i = 1:k
```

Comment:
I have a doubt, as you told, we can get new differences by `(u - uprev)*dt/dt_n-1` (which is equivalent to step changing scheme for order one) it's like linear relation, so will It be good for higher order method because `QNDF` is not doing good.
in paper, they wrote -> 
The ode15s is not allowed to change either the order or the step size until a minimum of
(k + 2) consecutive steps are given using the same order formula and the same step size.

Reply:
Those are somewhat unrelated. Using divided differences updates you can change the stepsize while retaining high order, and that should be used. But, note that changing the stepsize then requires the interpolation to the new grid. This interpolation reduces the stability, so it is smart to minimize the number of dt changes at high order. So putting the two together: allow order changes but require a few fixed timesteps first.